azure/services/async/async.go

```diff
@@ -140,6 +140,13 @@ func (s *Service) CreateOrUpdateResource(ctx context.Context, spec azure.Resourc
 	log.V(2).Info(fmt.Sprintf("%sing resource", logMessageVerbPrefix), "service", serviceName, "resource", resourceName, "resourceGroup", rgName)
 	result, sdkFuture, err := s.Creator.CreateOrUpdateAsync(ctx, spec, parameters)
 	errWrapped := errors.Wrapf(err, fmt.Sprintf("failed to %se resource %s/%s (service: %s)", logMessageVerbPrefix, rgName, resourceName, serviceName))
+
+	// If it is an intermittent failure with context deadline exceeded or canceled as the reconciler could not complete
+	// in the max amount of time, mark it as a transient error and return.
+	if err != nil && ContextDeadlineExceededOrCanceled(ctx) {
```

Comment:
in this case does err == ctx.Err() ? if so we could just pass err directly

Reply:
Passing err directly will not help to figure out if the context has exceeded or got canceled. We need the `ctx` object to figure that out. 
But I think, I can omit checking the error and just that function can check if ctx.Err() is nil or not. 
I will make that change.